clab/config.go

```diff
@@ -211,6 +211,7 @@ func (c *CLab) createNodeCfg(nodeName string, nodeDef *types.NodeDefinition, idx
 		Sandbox:         c.Config.Topology.GetNodeSandbox(nodeName),
 		Kernel:          c.Config.Topology.GetNodeKernel(nodeName),
 		Runtime:         c.Config.Topology.GetNodeRuntime(nodeName),
+		MaxWorkers:      c.Config.Topology.GetNodeMaxWorkers(nodeName),
```

Comment:
Do we want to make MaxWorkers to be a part of node config?
To me the number of max workers is a runtime's thing, not node's thing. A node is a single thing, it seems it doesn't add up having workers under a node.

If I think about the workers constraints, I'd think along the following lines:

1. build a `map[string][]nodes.Node` of nodes which has two keys
     1. `concurrent` -  this will have a list of nodes which are not artificially constrained (docker, containerd). The max workers will follow the current rule: no constraints, until set via flag
    2. `sequential` - this will have a list of nodes which are associated with runtimes which can't handle concurrent deployment (ignite)
1. then this map can be passed to `c.CreateNodes` and processed there

Maybe I am missing something here which makes this approach not feasible?

Reply:
My thinking was to remove any node-specific logic from `clab`. The property of concurrency is node-dependent so it's natural for it to come out of node's config and not get decided in `clab`.
Another issue is that if `cvx` is running in docker runtime, it should have to be serialized, so we can override this default behaviour with `maxworker` field of the node.

Reply:
but isn't it a runtime property still? as you mention, the cvx running in docker shouldn't be serialized, so it is the runtime that is a deciding factor here, not the node per-se. I.e. if we say for `cvx` node that the runtime is `docker`, then this is enough to make a decision to not serialize the deployment of the nodes.

another example, by exposing `maxworker` it is possible to do

```yaml
nodes:
  n1:
    kind: linux
    max-workers: 2
  n2:
    kind: linux
    max-workers: 4
```

but this doesn't make sense?

Reply:
yeah, your example is ambiguous but can be deterministic, e.g. if we sort all nodes first and say that the first (or last) value always wins. 

So, if I understood you correctly, this is what you're suggesting:
```go
var nodeWorkers map[string][]nodes.Node
for _, n := range c.Nodes {
  if n.GetRuntime() == runtime.IgniteRuntime {
    nodeWorkers[1] = append(nodeWorkers[1], n)
    continue
  }
  nodeWorkers[0] = append(nodeWorkers[0], n)
}
```

Reply:
I agree this is runtime dependent hence should not be any kind of Node property.

Reply:
I think it doesn't have to be a node param.
a single `map[string]struct{}`, built after nodes are initialized, that contains the nodes names that need to be deployed sequentially is enough.

Deploy the nodes from the built map sequentially, then the ones that are not in the map using the defined number of workers.